compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -934,7 +934,7 @@ public void visit(BLangRecordTypeNode recordTypeNode, AnalyzerData data) {
         for (BLangSimpleVariable field : recordFields) {
             if (field.flagSet.contains(Flag.READONLY)) {
                 handleReadOnlyField(isRecordType, fields, field, data);
-            } else {
+            } else if (field.getBType().getKind() != TypeKind.NEVER) {
```

Comment:
Won't work for effectively `never` structures?

A better check would be to check if assignable to `never` IMO. Alternatively, could use `Types#isNeverTypeOrStructureTypeWithARequiredNeverMember` too, but the assignability check would be better.

Reply:
```ballerina
type NeverRecord record {|
    record {|
        never a;
    |} x?;
|};

public function main() {
    NeverRecord a = {};
    readonly b = a; // still an error
}
```

Reply:
And 

```ballerina
type NeverRecord record {|
    never|never x?;
|};
```